presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -266,10 +270,12 @@ public long getPeakTotalMemoryInBytes()
         return peakTotalMemory.get();
     }
 
-    public void updateMemoryUsage(long currentUserMemoryValue, long currentSystemMemory)
+    public void updateMemoryUsage(long deltaUserMemoryInBytes, long deltaTotalMemoryInBytes)
     {
-        peakUserMemory.accumulateAndGet(currentUserMemoryValue, Math::max);
-        peakTotalMemory.accumulateAndGet(currentUserMemoryValue + currentSystemMemory, Math::max);
+        long currentUserMemoryValue = currentUserMemory.addAndGet(deltaUserMemoryInBytes);
```

Comment:
`currentXXMemoryValue` is a weird name; maybe we can just inline these two variables. Or use `currentUserMemory.get()` below.